src/everest/config/everest_config.py

```diff
@@ -452,15 +452,18 @@ def validate_variable_name_match_well_name(self) -> Self:  # pylint: disable=E02
         wells = self.wells
         if controls is None or wells is None:
             return self
-        well_names = [w.name for w in wells]
+        well_names = {w.name for w in wells}
         if not well_names:
             return self
         for c in controls:
-            if c.type == "generic_control":
-                continue
-            for v in c.variables:
-                if v.name not in well_names:
-                    raise ValueError("Variable name does not match any well name")
+            variable_names = {v.name for v in c.variables}
+            if len(variable_names.intersection(well_names)) not in {
+                0,
+                len(variable_names),
+            }:
+                raise ValueError(
+                    "Variable names should either all or none match well names"
+                )
```

Comment:
So this case would not be allowed anymore: A control group where most variable names are arbitrary, but one or more happen to be well names?

If so, it seems an arbitrary limitation, somebody may have a case that breaks it.

Reply:
Yes, that will no longer be allowed. 

We could remove this arbitrary limitation but once the control type filed is removed we can no longer enforce any constraint on the well name and variable names. So as far as can see there we have the following options:

1. We keep the control type. 
2. We remove the control type and this validation 
3. We add this arbitrary limitation that might break some cases where there are `well_controls` and `generic_controls` that happen to have some of the variable names match the well names.

 There could be some other options I am not considering at the moment, but I am open to suggestions.